common/src/main/java/com/wynntils/screens/widgets/QuestButton.java

```diff
@@ -0,0 +1,143 @@
+import org.lwjgl.glfw.GLFW;
+
+public class QuestButton extends AbstractButton {
+    private static final CustomColor BUTTON_COLOR = new CustomColor(181, 174, 151);
+    private static final CustomColor BUTTON_COLOR_HOVERED = new CustomColor(121, 116, 101);
+    private static final CustomColor TRACKED_BUTTON_COLOR = new CustomColor(176, 197, 148);
+    private static final CustomColor TRACKED_BUTTON_COLOR_HOVERED = new CustomColor(126, 211, 106);
+    private final QuestInfo questInfo;
+    private final WynntilsQuestBookScreen questBookScreen;
+
+    public QuestButton(int x, int y, int width, int height, QuestInfo questInfo, WynntilsQuestBookScreen screen) {
+        super(x, y, width, height, new TextComponent("Quest Button"));
+        this.questInfo = questInfo;
+        this.questBookScreen = screen;
+    }
+
+    @Override
+    public void renderButton(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        CustomColor backgroundColor = questBookScreen.getTracked() == this.questInfo
+                ? (this.isHovered ? TRACKED_BUTTON_COLOR_HOVERED : TRACKED_BUTTON_COLOR)
+                : (this.isHovered ? BUTTON_COLOR_HOVERED : BUTTON_COLOR);
+
+        RenderUtils.drawRect(poseStack, backgroundColor, this.x, this.y, 0, this.width, this.height);
+
+        int maxTextWidth = this.width - 10 - 11;
+        FontRenderer.getInstance()
+                .renderText(
+                        poseStack,
+                        StringUtils.getMaxFittingText(
+                                questInfo.getName(),
+                                maxTextWidth,
+                                FontRenderer.getInstance().getFont()),
+                        this.x + 14,
+                        this.y + 1,
+                        0,
+                        CommonColors.BLACK,
+                        HorizontalAlignment.Left,
+                        VerticalAlignment.Top,
+                        FontRenderer.TextShadow.NONE);
+
+        Texture stateTexture =
+                switch (questInfo.getStatus()) {
+                    case STARTED -> Texture.QUEST_STARTED;
+                    case COMPLETED -> Texture.QUEST_FINISHED;
+                    case CAN_START -> Texture.QUEST_CAN_START;
+                    case CANNOT_START -> Texture.QUEST_CANNOT_START;
+                };
+
+        RenderUtils.drawTexturedRect(
+                poseStack,
+                stateTexture.resource(),
+                this.x + 1,
+                this.y + 1,
+                stateTexture.width(),
+                stateTexture.height(),
+                stateTexture.width(),
+                stateTexture.height());
+    }
+
+    // Not called
+    @Override
+    public void onPress() {}
+
+    @Override
+    public boolean mouseClicked(double mouseX, double mouseY, int button) {
+        if (button == GLFW.GLFW_MOUSE_BUTTON_LEFT) {
+            if (this.questInfo.getStatus() != QuestStatus.CANNOT_START) {
```

Comment:
Maybe break out this (and the "open wiki") part to separate methods, so `mouseClicked` is just a small dispatcher.